### PR TITLE
API 1.31

### PR DESCRIPTION
**Pull request by onabiull**

### CODE REVIEW

File: samples/sample_multi_transcode/src/transcode_utils.cpp [hunk 1]
```diff
@@ -1187,6 +1187,16 @@ mfxStatus ParseAdditionalParams(msdk_char *argv[], mfxU32 argc, mfxU32& i, Trans
     {
         InputParams.nVuiNalHrdParameters = MFX_CODINGOPTION_OFF;
     }
+#if (defined(_WIN32) || defined(_WIN64)) && (MFX_VERSION >= 1031)
```
Comment (dvrogozh) [inline samples/sample_multi_transcode/src/transcode_utils.cpp]: why did you move the block? you distant windows/linux adapter options from each other...

File: samples/sample_encode/src/pipeline_encode.cpp [hunk 2]
```diff
@@ -1290,7 +1290,7 @@ mfxStatus CEncodingPipeline::InitFileWriters(sInputParams *pParams)
     return sts;
 }
 
-#if (defined(_WIN64) || defined(_WIN32)) && (MFX_VERSION >= MFX_VERSION_NEXT)
+#if (defined(_WIN64) || defined(_WIN32)) && (MFX_VERSION >= 1031)
```
Comment (dvrogozh) [inline samples/sample_encode/src/pipeline_encode.cpp]: why version check is here?! Just few lines below in a function you have #if (MFX_VERSION >= 1027) - it does not make sense!

File: samples/sample_multi_transcode/src/transcode_utils.cpp (hunk 1 above)
Comment (onabiull) [reply samples/sample_multi_transcode/src/transcode_utils.cpp]: On windows we hit compiler limit here. Too long if/else chain lead to compiler error. This function specifically designed to workaround this limitation.

File: samples/sample_encode/src/pipeline_encode.cpp (hunk 2 above)
Comment (onabiull) [reply samples/sample_encode/src/pipeline_encode.cpp]: To split 12-bit enabling from discrete adapters.
